### PR TITLE
feat: implement `random init` subroutine

### PR DESCRIPTION
**Pull request by parth121101**
Towards: #4169

### CODE REVIEW

**Comment by parth121101** [comment]
Still some changes is required before we merge this PR.
Here are the results for this changes:

```
cat a.f90
program test
    implicit none

    real :: x(1, 3)
    x = 0
    call random_init(.true., .true.)
    call random_number(x(1, 2))
    call random_init(.true., .true.)
    call random_number(x(1, 3))
    print *, x

    if (abs(x(1, 1) - 0.0) > 1e-5) error stop
    if (abs(x(1, 2) - 0.0) < 1e-8) error stop
    ! The two values must be the same, due to `random_init()` above
    if (x(1, 2) /= x(1, 3)) error stop

end program%                       
                                                                                                                                     
% lfortran a.f90
0.00000000e+00 8.61170113e-01 8.61170113e-01 
```

File: src/libasr/runtime/lfortran_intrinsics.c [hunk 1]
```diff
@@ -2200,6 +2201,20 @@ LFORTRAN_API double _lfortran_dp_rand_num() {
...
 }
 
+
+LFORTRAN_API int _lfortran_random_init(bool repeatable, bool image_distinct) {
+    if (repeatable) {
+        if (image_distinct) {
+            srand(time(NULL) + getpid());
+        } else {
+            srand(42);
+        }
```
Comment (certik) [inline src/libasr/runtime/lfortran_intrinsics.c:2211]: For now I would ignore `image_distinct`, that only is used for parallel co-array code, which we do not support yet. Just return `srand(42)` here. Also, let's return the same seed as GFortran does (0?).

File: src/libasr/runtime/lfortran_intrinsics.c [hunk 2]
```diff
@@ -9,6 +9,7 @@
 #include <float.h>
 #include <limits.h>
 #include <ctype.h>
+#include <unistd.h>
```
Comment (certik) [inline src/libasr/runtime/lfortran_intrinsics.c:12]: This line fails on Windows:
```
2024-07-05T16:34:59.3342664Z D:\a\lfortran\lfortran\lfortran-0.37.0-56-g25f20c484\src\libasr\runtime\lfortran_intrinsics.c(12): fatal error C1083: Cannot open include file: 'unistd.h': No such file or directory
```

**Review by certik — APPROVED** [review]
Thanks! I think this is a good start. We can iterate on this later as we discover issues.

**Pull request merged**